### PR TITLE
Strawman compatibility categories

### PR DESCRIPTION
**Pull request by dthaler**
Signed-off-by: Dave Thaler <dthaler@armidalesoftware.com>

### CODE REVIEW

**Comment by jamestanner45** [comment]
Looks good to me.

**Comment by albertemmerich** [comment]
For Tree Compatibility level 2 we should have 
`<<EVENT_DETAIL>>.NOTE`

**Comment by albertemmerich** [comment]
Will there be a compatibility level 3, with full spec covered?

**Comment by dthaler** [comment]
> For Tree Compatibility level 2 we should have `<<EVENT_DETAIL>>.NOTE`

Agree, added.

File: _pages/compatibility.md [hunk 1]
```diff
@@ -9,26 +9,68 @@ sidebar:
 
 ## Compatibility Requirements
 
+* READ/WRITE FAMILYSEARCH GEDCOM FILE
+> The vendor needs to demonstrate the ability to both read and write a FamilySearch GEDCOM file.  The [maximal70.ged](..\testfiles\gedcom70\maximal70.ged) file can be used to determine a [compatbility score](https://magikeygedcomconverter.azurewebsites.net/Compatability).
+
 * READ FAMILYSEARCH GEDCOM FILE
-> The vendor needs to demonstrate the ability to read a sample FamilySearch GEDCOM file and display the contents in their own environment.
+> The vendor only needs to demonstrate the ability to read the [maximal70.ged](..\testfiles\gedcom70\maximal70.ged) file and display the contents in their own environment.
```
Comment (tychonievich) [inline _pages/compatibility.md:16]:
```suggestion
> The vendor only needs to demonstrate the ability to read the [maximal70.ged](../testfiles/gedcom70/maximal70.ged) file and display the contents in their own environment.
```

**Comment by tychonievich** [comment]
I think we are missing some important components. 

- There ought to be some styling compatibility (NOTE.MIME, SOUR.TEXT.MIME, other MIME if SNOTE and SNOTE.TRAN supported)
- There should be a logging compatibility that always sets CREA and CHAN dates appropriately
- There should be an extension compatibility that can retag conflicting documented extension tags upon import and present relocated standard structures in the user interface
- There should be at least one "full compatibility" level, and probably two:
   - One that lets users read, modify, and create any standards-compliant file
   - One that lets users read any standards-compliant file but may omit tools to create some (e.g. not permitting duplicate data, moving single-use SNOTE to NOTE, limiting ALIA use to particular situations, normalizing name parts, enforcing recommendations as well as requirements, etc.)

I do not support publishing this unless a full compatibility level is included; otherwise it looks like we are saying "most of the standard doesn't matter."

**Review by tychonievich — CHANGES_REQUESTED** [review]

File: _pages/compatibility.md [hunk 2]
```diff
@@ -9,26 +9,68 @@ sidebar:
 
 ## Compatibility Requirements
 
+* READ/WRITE FAMILYSEARCH GEDCOM FILE
+> The vendor needs to demonstrate the ability to both read and write a FamilySearch GEDCOM file.  The [maximal70.ged](..\testfiles\gedcom70\maximal70.ged) file can be used to determine a [compatbility score](https://magikeygedcomconverter.azurewebsites.net/Compatability).
+
 * READ FAMILYSEARCH GEDCOM FILE
-> The vendor needs to demonstrate the ability to read a sample FamilySearch GEDCOM file and display the contents in their own environment.
+> The vendor only needs to demonstrate the ability to read the [maximal70.ged](../testfiles/gedcom70/maximal70.ged) file and display the contents in their own environment.
```
Comment (tychonievich) [inline _pages/compatibility.md:16]:
```suggestion
> The vendor needs to demonstrate the ability to read the [maximal70.ged](../testfiles/gedcom70/maximal70.ged) file and display the contents in their own environment.
```

File: _pages/compatibility.md [hunk 3]
```diff
@@ -9,26 +9,68 @@ sidebar:
...
 
 
 * WRITE FAMILYSEARCH GEDCOM FILE
-> The vendor needs to demonstrate the ability to write a sample FamilySearch GEDCOM file and prove that it can be read by a FamilySearch GEDCOM Compatible product.
+> The vendor only needs to demonstrate the ability to write a sample FamilySearch GEDCOM file and prove that it is valid FamilySearch GEDCOM 7. For example, the [GEDCOM validator](http://ged-inline.elasticbeanstalk.com/) can be used, although it is still in beta.
```
Comment (tychonievich) [inline _pages/compatibility.md:20]:
```suggestion
> The vendor needs to demonstrate the ability to write a sample FamilySearch GEDCOM file and prove that it is valid FamilySearch GEDCOM 7. For example, the [GEDCOM validator](http://ged-inline.elasticbeanstalk.com/) can be used, although it is still in beta.
```

Comment (tychonievich) [inline _pages/compatibility.md:20]:
```suggestion
> The vendor only needs to demonstrate the ability to write a sample FamilySearch GEDCOM 7 file and that any FamilySearch GEDCOM 7 files they write are conformant with the FamilySearch GEDCOM 7 specification. For example, the [GEDCOM validator](http://ged-inline.elasticbeanstalk.com/) can be used to check validity, although it is still in beta.
```

**Review by tychonievich — CHANGES_REQUESTED** [review]
Some typos and wording changes

**Comment by albertemmerich** [comment]
There are applications which import anything and let at export all parts unmodified they do not understand. These applications would get a very high score if we only test import of compliant files and then export.
Should we add a test for the reaction of applications, if they import a file which is not compliant? If they do not react and do not eliminate any deviations to standard rules, they cannot be called compatibel: They export the deviations and by that break the standard rules.

**Comment by dthaler** [comment]
> There are applications which import anything and let at export all parts unmodified they do not understand. These applications would get a very high score if we only test import of compliant files and then export. Should we add a test for the reaction of applications, if they import a file which is not compliant? If they do not react and do not eliminate any deviations to standard rules, they cannot be called compatibel: They export the deviations and by that break the standard rules.

Agree.  We had this under Write but not Read/Write.  I've now added it there too for consistency.

**Comment by tychonievich** [comment]
At a high level, ignoring the specific tags supported, I think we have:

**Write-compliant**:
Any 7.0 file exported by the application is valid GEDCOM

**Read-compliant**:
Imposes no import limitations not present in the GEDCOM specification

**Delta-compliant**:
An import-modify-export introduces no new errors beyond those present in the imported file

**Comment by dthaler** [comment]
Albert reports that John Cardinal has an invalid file (assess.ged) at https://gedcomassessment.com/en/index.htm that may be useful for applications wanting to test with an invalid GEDCOM 5.5.1 file.

**Review by tychonievich — APPROVED** [review]

**Comment by jamestanner45** [comment]
Albert is right. I agree. Garbage in garbage out.

James

**Pull request merged**